### PR TITLE
fix: get plugin app vsn from field `rel_apps`

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes [EMQX-12479](https://emqx.atlassian.net/browse/EMQX-12479)

Release version: v/e5.7.1

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

[EMQX-12479]: https://emqx.atlassian.net/browse/EMQX-12479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/emqx_plugins/src/emqx_plugins.erl [hunk 1]
```diff
@@ -136,6 +136,24 @@ parse_name_vsn(NameVsn) when is_list(NameVsn) ->
...
 
+app_dir(AppName, Apps) ->
+    case
+        lists:filtermap(
+            fun(AppNameVsn) ->
+                case string:prefix(AppNameVsn, AppName) of
+                    nomatch -> false;
+                    _Remaing -> {true, AppNameVsn}
+                end
+            end,
```
Comment (thalesmg) [inline apps/emqx_plugins/src/emqx_plugins.erl:147]: Micro nit:

```suggestion
        lists:filter(
            fun(AppNameVsn) ->
                nomatch =/= string:prefix(AppNameVsn, AppName)
            end,
```

**Review by thalesmg — COMMENTED** [review]
Maybe add a test?

File: apps/emqx_plugins/src/emqx_plugins.erl [hunk 2]
```diff
@@ -1372,12 +1390,14 @@ plugin_dir(NameVsn) ->
...
-            wrap_to_list(filename:join([plugin_dir(NameVsn), AppDir, "priv"]));
-        _ ->
-            wrap_to_list(filename:join([install_dir(), NameVsn, "priv"]))
+    maybe
+        {ok, #{<<"name">> := Name, <<"rel_apps">> := Apps}} ?=
+            read_plugin_info(NameVsn, #{fill_readme => false}),
+        {ok, AppDir} ?= app_dir(Name, Apps),
+        wrap_to_list(filename:join([plugin_dir(NameVsn), AppDir, "priv"]))
+    else
+        %% Otherwise assume the priv directory is under the plugin root directory
```
Comment (zmstone) [inline apps/emqx_plugins/src/emqx_plugins.erl:1399]: I am not sure why this was added in the first place.
doesn't `priv` dir always reside in`.../app_name-x.y.z/` ?

Comment (JimMoen) [reply apps/emqx_plugins/src/emqx_plugins.erl:1399]: We assume the `rel_apps` included an application named same with release.
But if the application doesn't, plugin author should place an priv dir into project root.

See details below:
`rel_apps` does not have an application `otp_ts_ui` included.

<details>

```
$ bat otp_ts_ui-1.1.0/release.json
───────┬────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: otp_ts_ui-1.1.0/release.json
───────┼────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ {
   2   │     "name": "otp_ts_ui",
   3   │     "description": "Another amazing EMQX plugin.",
   4   │     "authors": [
   5   │         "JimMoen"
   6   │     ],
   7   │     "builder": {
   8   │         "name": "JimMoen",
   9   │         "contact": "LnJimMoen@outlook.com",
  10   │         "website": "http://example.com"
  11   │     },
  12   │     "repo": "https://github.com/emqx/emqx-plugin-template",
  13   │     "functionality": [
  14   │         "Demo"
  15   │     ],
  16   │     "compatibility": {
  17   │         "emqx": "~> 5.7"
  18   │     },
  19   │     "git_ref": "58e7c43a8e41a9fb62c5d12082b351f087b308f7",
  20   │     "built_on_otp_release": "26",
  21   │     "git_commit_or_build_date": "2024-05-29",
  22   │     "metadata_vsn": "0.1.0",
  23   │     "rel_apps": [
  24   │         "test_app-0.2.0",
  25   │         "map_sets-1.1.0"
  26   │     ],
  27   │     "rel_vsn": "1.1.0",
  28   │     "with_config_schema": false
  29   │ }
───────┴────────────────────────────────────────────────────────────────────────────────────────────────

$ tree                            
.
├── otp_ts_ui-1.1.0
│   ├── map_sets-1.1.0
│   │   └── ebin
│   │       ├── map_sets.app
│   │       └── map_sets.beam
│   ├── README.md
│   ├── release.json
│   └── test_app-0.2.0
│       └── ebin
│           ├── otp_ts_ui_app.beam
│           ├── otp_ts_ui.beam
│           ├── otp_ts_ui_cli.beam
│           ├── otp_ts_ui_sup.beam
│           └── test_app.app
├── otp_ts_ui-1.1.0.sha256
└── otp_ts_ui-1.1.0.tar.gz

6 directories, 11 files
```

</details>

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2437)

**Pull request merged**